src/client/datascience/commands/commandRegistry.ts

```diff
@@ -496,11 +498,12 @@ export class CommandRegistry implements IDisposable {
     }
     private async onVariablePanelShowDataViewerRequest(request: IShowDataViewerFromVariablePanel) {
         sendTelemetryEvent(EventName.OPEN_DATAVIEWER_FROM_VARIABLE_WINDOW_REQUEST);
-        if (this.debugService.activeDebugSession) {
-            const jupyterVariable = convertDebugProtocolVariableToIJupyterVariable(
-                request.variable as DebugProtocol.Variable
-            );
+        if (this.debugService.activeDebugSession && this.variableProvider.getFullVariable) {
             try {
+                const variable = convertDebugProtocolVariableToIJupyterVariable(
+                    request.variable as DebugProtocol.Variable
+                );
+                const jupyterVariable = await this.variableProvider.getFullVariable(variable);
```

Comment:
What we were doing previously (superficial conversion from `DebugProtocol.Variable` to `IJupyterVariable`) was sufficient for what we shipped in Jan, but isn't enough for slicing, because for slicing we need to know the `shape` and `type`. Those weren't being filled out for DebuggerVariables before because it wasn't necessary until now.